File: src/main/java/br/com/alura/forum/repository/UserRepository.java
package br.com.alura.forum.repository;

import java.util.Optional;

import br.com.alura.forum.model.User;
import org.springframework.data.repository.Repository;

public interface UserRepository extends Repository<User, Long>{

	Optional<User> findByEmail(String email);
	Optional<User> findById(Long userId);
	User save(User user);
	
}

File: src/main/kotlin/br/com/alura/forum/repository/UserRepository.kt
package br.com.alura.forum.repository

import br.com.alura.forum.model.User
import org.springframework.data.repository.Repository

interface UserRepository : Repository<User, Long> {
    fun findByEmail(email: String?): User?
    fun findById(userId: Long): User?
    fun save(user: User): User
}